### PR TITLE
setup Swagger.

### PR DESCRIPTION
**Pull request by nideyidayida**

### CODE REVIEW

**Review by MingshiPeng — CHANGES_REQUESTED** [review]

File: app/__init__.py [hunk 1]
```diff
@@ -9,3 +9,19 @@
 db = SQLAlchemy(app)
 
 from app import server, models
+from flasgger import Swagger
+
+# Configurue Swagger
+app.config['SWAGGER'] = {
```
[the remark this answers is not in the document]
Comment (jockyxu1122) [reply app/__init__.py]: addressed

File: app/server.py [hunk 2]
```diff
@@ -76,6 +76,7 @@ def index():
     return jsonify(
         name='Inventory Management REST API Service',
         version='1.0',
+        docs=request.base_url + 'apidocs/index.html',
```
[the remark this answers is not in the document]
Comment (jockyxu1122) [reply app/server.py]: It's the link to Swagger doc.

File: config.py [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
 from app.vcap_services import get_database_uri
+from app import app
+from flasgger import Swagger
 
 SQLALCHEMY_DATABASE_URI = get_database_uri()
 SQLALCHEMY_TRACK_MODIFICATIONS = False
 LOGGING_LEVEL = logging.INFO
+
+# Configurue Swagger
+app.config['SWAGGER'] = {
```
Comment (MingshiPeng) [inline config.py]: Use syntax similar to line 6, 7, 8, e.g. 
SWAGGER = {...}
instead of 
app.config['SWAGGER'] = {...}

File: config.py [hunk 4]
```diff
@@ -1,6 +1,23 @@
 import logging
 from app.vcap_services import get_database_uri
+from app import app
```
Comment (MingshiPeng) [inline config.py]: Not needed.
Refer to comment in line 11.

File: app/__init__.py [hunk 5]
```diff
@@ -8,20 +8,4 @@
...
-            "title": "Inventory",
-            "description": "This is an Inventory server.",
-            "endpoint": 'v1_spec',
-            "route": '/v1/spec'
-        }
-    ]
-}
-Swagger(app)
\ No newline at end of file
+from app import server, models
```
Comment (MingshiPeng) [inline app/__init__.py]: new line.

File: config.py [hunk 6]
```diff
@@ -1,6 +1,23 @@
...
+from flasgger import Swagger
```
Comment (MingshiPeng) [inline config.py]: Not needed.

File: config.py [hunk 7]
```diff
@@ -1,6 +1,23 @@
...
+        {
+            "version": "1.0.0",
+            "title": "Inventory",
+            "description": "This is an Inventory server.",
+            "endpoint": 'v1_spec',
+            "route": '/v1/spec'
+        }
+    ]
+}
+Swagger(app)
```
Comment (MingshiPeng) [inline config.py]: This line should be in __init__.py after line
app.config.from_object('config')
config.py contains configurations only and should not do any actual work or registration.

**Review by MingshiPeng — CHANGES_REQUESTED** [review]

File: config.py (hunk 3 above)
Comment (jockyxu1122) [reply config.py]: addressed

File: config.py (hunk 4 above)
Comment (jockyxu1122) [reply config.py]: addressed

File: app/__init__.py (hunk 5 above)
Comment (jockyxu1122) [reply app/__init__.py]: addressed

File: config.py (hunk 6 above)
Comment (jockyxu1122) [reply config.py]: addressed

File: config.py (hunk 7 above)
Comment (jockyxu1122) [reply config.py]: addressed

**Review by MingshiPeng — APPROVED** [review]
Ship it!

**Comment by majia67** [comment]
IBM Cloud [toolchain](https://console.ng.bluemix.net/devops/toolchains/d4dd8efb-da17-4b0c-b4e6-5599ec73dd60?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18"): [Delivery Pipeline](https://console.ng.bluemix.net/devops/pipelines/c71be43a-8963-4052-b97c-5da9b99de3f9?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18") deployed [inventory](https://github.com/NYUDevOpsInventory/inventory/releases/tag/deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-dev-20180424-184916 "deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-dev-20180424-184916") to [dev](https://console.bluemix.net/apps/1db994d8-c4f1-4076-85c0-7ff2674244e5?env_id=ibm:yp:us-south "ibm:yp:us-south:NYU_Devops_Inventory_Org:dev"), including commit 4907b21ad4376b03bfd247828fdb0496ff6248e9

**Comment by majia67** [comment]
IBM Cloud [toolchain](https://console.ng.bluemix.net/devops/toolchains/d4dd8efb-da17-4b0c-b4e6-5599ec73dd60?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18"): [Delivery Pipeline](https://console.ng.bluemix.net/devops/pipelines/c71be43a-8963-4052-b97c-5da9b99de3f9?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18") deployed [inventory](https://github.com/NYUDevOpsInventory/inventory/releases/tag/deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-prod-20180501-183630 "deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-prod-20180501-183630") to [prod](https://console.bluemix.net/apps/e25e653b-a50f-4831-b93e-d8f9f484a484?env_id=ibm:yp:us-south "ibm:yp:us-south:NYU_Devops_Inventory_Org:prod"), including commit 4907b21ad4376b03bfd247828fdb0496ff6248e9

**Pull request merged**